src/streamlink/plugins/pandalive.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?pandalive\.co\.kr/"
+))
+class Pandalive(Plugin):
+    API_URL = "https://api.pandalive.co.kr/v1/live/play"
+
+    _room_id_re = re.compile(r"roomid\s*=\s*String\.fromCharCode\((.*)\)")
+
+    def _get_media_code(self):
+        root = self.session.http.get(self.url, schema=validate.Schema(
+            validate.parse_html(),
+        ))
+
+        try:
+            schema = validate.Schema(
+                validate.xml_xpath_string(".//script[contains(text(), 'roomid')]/text()"),
+                str,
+                validate.transform(self._room_id_re.search),
+                validate.get(1),
+                str,
+                validate.transform(lambda s: "".join(map(lambda c: chr(int(c)), s.split(",")))),
+            )
+            return schema.validate(root)
+        except PluginError:
+            pass
```

Comment:
If regex.search returns None, this will raise a TypeError. And getting a capture group will always be a string, so this secondary string type validation is unnecessary.

You also don't need to split up the validation schema if you're not re-using the parsed HTML root element.